### PR TITLE
fix(Core/Totems): Totems should not fall to the depth in water.

### LINKED ISSUE #6562: Shaman totems placed while swimming will sink underwater
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [ ] Alliance
- [x] Horde

##### CONTENT PHASE:

- [x] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

When placing totems while swimming underwater, they will sink until hitting something (land, rock, etc).

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Totems should float near player model when placed, they should not sink.

##### SOURCE:

Retail Classic WoW: https://youtu.be/cHSL9yvrnqU?t=456

##### STEPS TO REPRODUCE THE PROBLEM:

1.  As a shaman, swim underwater
2.  Place any totem
3.  Totem sinks until it reaches the water's floor or collides with something in its vertical path, rock, structure, etc.

##### EXTRA NOTES:

Originally reported https://github.com/chromiecraft/chromiecraft/issues/997

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/34da0cda5195dede48cb0406b23248330d6249cb

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/99451417-shaman-totems-placed-while-swimming-will-sink-underwater?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
Fixed #6562

## Changes Proposed:
-  Totems should not fall to the depth in water

## Issues Addressed:

- Closes #6562
- Closes https://github.com/chromiecraft/chromiecraft/issues/997

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

1. As a shaman, swim underwater
2. Place any totem
3. Totem sinks until it reaches the water's floor or collides with something in its vertical path, rock, structure, etc.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by pklloveyou** [comment]
It won't sink when you're underwater. It will sink to the bottom when you're on the surface
image
image

**Comment by UltraNix** [comment]
But this not a bottom...

**Comment by pklloveyou** [comment]
ok

**Comment by pklloveyou** [comment]
ok
Tested

File: src/server/game/Entities/Totem/Totem.cpp [hunk 1]
```diff
@@ -88,7 +88,8 @@ void Totem::InitSummon()
         GetOwner()->CastSpell(this, 6277, true);
     }
 
-    this->GetMotionMaster()->MoveFall();
+    if (!IsInWater())
+        GetMotionMaster()->MoveFall();
```
Comment (Helias) [inline src/server/game/Entities/Totem/Totem.cpp:92]:
```suggestion
    if (!IsInWater())
    {
        GetMotionMaster()->MoveFall();
    }
```

**Review by Helias — APPROVED** [review]

**Pull request merged**